mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -229,6 +233,120 @@ impl TendermintCoin {
             .parse()
             .map_to_mm(|e| TendermintCoinRpcError::InvalidResponse(format!("balance is not u64, err {}", e)))
     }
+
+    #[allow(dead_code)]
+    fn gen_create_htlc_tx(
+        &self,
+        base_denom: Denom,
+        to: &AccountId,
+        amount: cosmrs::Decimal,
+        secret_hash: &[u8],
+        time_lock: u64,
+    ) -> MmResult<IrisHtlc, TxMarshalingErr> {
+        let timestamp = get_utc_timestamp() as u64;
+        let mut hash_lock_hash = vec![];
+        hash_lock_hash.extend_from_slice(secret_hash);
+        hash_lock_hash.extend_from_slice(&timestamp.to_be_bytes());
+
+        let amount = vec![Coin {
+            denom: self.denom.clone(),
+            amount,
+        }];
+
+        // Needs to be sorted if cointains multiple coins
+        // amount.sort();
+
+        // << BEGIN HTLC id calculation
+        let mut htlc_id = vec![];
+        htlc_id.extend_from_slice(sha256(&hash_lock_hash).as_slice());
+        htlc_id.extend_from_slice(&self.account_id.to_bytes());
+        htlc_id.extend_from_slice(&to.to_bytes());
+        let mut coins_string = String::new();
+        for (index, item) in amount.iter().enumerate() {
+            if index == 0 {
+                coins_string = format!("{}{}{}", coins_string, item.amount, item.denom);
+            } else {
+                coins_string = format!(",{}{}{}", coins_string, item.amount, item.denom);
+            }
+        }
+        htlc_id.extend_from_slice(coins_string.as_bytes());
+        let htlc_id = sha256(&htlc_id).to_string().to_uppercase();
+        // >> END HTLC id calculation
+
+        let msg_payload = MsgCreateHtlc {
+            sender: self.account_id.clone(),
+            to: to.clone(),
+            receiver_on_other_chain: "".to_string(),
```

Comment:
Could you please provide more details on when these `receiver/sender_on_other_chain` fields should have non-default values?

Reply:
I think it's never required but it can be good for state management for swapping coins cross chains.

For example, if we swap ATOM with BTC, we can use/validate the swap addresses directly from the transaction itself using `receiver/sender_on_other_chain`. At least this is how I understand it. There is no detailed information about them. I will ask IRIS team to validate if I am correct about this.